src/app.js

```diff
@@ -142,16 +142,19 @@ module.exports.run = function (worker) {
       query.fields = query.fields.split(',');
     }
 
-    prom.then((response) => {
+    prom.catch(err => {
+      log.error(String(err), {stacktrace: err.stack});
+      return {statusCode: 500, error: String(err)};
+    }).then((response) => {
       if ((typeof response !== 'object') ||
           (typeof response.statusCode !== 'number') ||
           (response.statusCode === 200 && !response.data) ||
           (response.statusCode !== 200 && !response.error)) {
-        log.warn('response is not wrapped in an envelope', {response: response});
+        log.error('response is not wrapped in an envelope', {response: response});
         response = {
-          statusCode: 200,
+          statusCode: 500,
           data: response,
-          errors: [{warning: 'missing envelope'}]
```

Comment:
Maybe we should create a random id (`uuid.v4()`) for all errors, log the error and id, and just send the id of the error to the user, in case they actually want to complain to somebody about it.